test/hisi_zip_test/test_lib.c

```diff
@@ -569,43 +569,75 @@ static void *poll_thread_func(void *arg)
 	pthread_exit(NULL);
 }
 
-int create_threads(struct hizip_test_info *info)
+int create_send_threads(struct hizip_test_info *info,
+			void *(*send_thread_func)(void *arg),
+			int num)
 {
 	pthread_attr_t attr;
-	int ret;
+	int i, ret;
 
-	count = 0;
-	info->thread_nums = 2;
-	info->threads = calloc(1, info->thread_nums * sizeof(pthread_t));
-	if (!info->threads)
+	info->send_tds = calloc(1, sizeof(pthread_t) * num);
+	if (!info->send_tds)
 		return -ENOMEM;
+	info->send_tnum = num;
 	pthread_attr_init(&attr);
 	pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_JOINABLE);
-	/* polling thread is the first one */
-	ret = pthread_create(&info->threads[0], &attr, poll_thread_func, info);
-	if (ret < 0) {
-		WD_ERR("fail to create poll thread (%d)\n", ret);
-		return ret;
+	for (i = 0; i < num; i++) {
+		ret = pthread_create(&info->send_tds[i], &attr,
+				     send_thread_func, info);
+		if (ret < 0) {
+			fprintf(stderr, "Fail to create send thread %d (%d)\n",
+				i, ret);
+			return ret;
+		}
 	}
-	ret = pthread_create(&info->threads[1], &attr, send_thread_func, info);
-	if (ret < 0)
-		return ret;
 	pthread_attr_destroy(&attr);
 	g_conf = &info->ctx_conf;
 	return 0;
```

Comment:
should add some error handle, like memory free, thread destroy, global variables unset